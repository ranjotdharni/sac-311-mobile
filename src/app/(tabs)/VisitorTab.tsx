import React from 'react';
import { View, ScrollView, Dimensions, Text, Pressable, StyleSheet, Image } from "react-native";
import {globalFont, global} from "../../customs"
import { router } from "expo-router";


export default function VisitorTab() {
    return(
        <View style={styles.pageWrapper}>
            <View style={{paddingTop:'10%'}}/>
            <View style={styles.tabName}>
                <Text style={styles.tabText}>Visitor Information</Text>
            </View>
            <Image style={styles.imageBanner} source={require('../../assets/png/visit.png')}/>
            <View style={styles.buttonMenu}>

                <View style={styles.buttonWrapper}>
                    <Text style={styles.buttonLabel}>Explore Sacramento</Text>
                    <Text style={styles.buttonDesc}>Find out more about what there is to do in Sacramento.</Text>
                    <Pressable style={styles.buttons} onPress={() => {router.push({pathname:'/(web)/WebView', params: {url: "https://www.visitcalifornia.com/places-to-visit/sacramento/"}})}}>
                        <Text style={styles.buttonText}>VisitCalifornia.com</Text>
                    </Pressable>
                </View>

                <View style={styles.buttonWrapper}>
                    <Text style={styles.buttonLabel}>Book Your Stay</Text>
                    <Text style={styles.buttonDesc}>Explore options booking a hotel, places to eat nearby, and more.</Text>
                    <Pressable style={styles.buttons} onPress={() => {router.push({pathname:'/(web)/WebView', params: {url: "https://www.visitsacramento.com/"}})}}>
                        <Text style={styles.buttonText}>VisitSacramento.com</Text>
                    </Pressable>
                </View>

                <View style={styles.buttonWrapper}>
                    <Text style={styles.buttonLabel}>Tripadvisor: Sacramento</Text>
                    <Text style={styles.buttonDesc}>Find hotels, vacation rentals, browse travel forums, and more on Tripadvisor.</Text>
                    <Pressable style={styles.buttons} onPress={() => {router.push({pathname:'/(web)/WebView', params: {url: "https://www.tripadvisor.com/Tourism-g32999-Sacramento_California-Vacations.html"}})}}>
                        <Text style={styles.buttonText}>Tripadvisor.com</Text>
                    </Pressable>
                </View>
            </View>
        </View>
    )
}

const styles = StyleSheet.create({
    pageWrapper:{
        width: '100%',
        height: '100%',
        backgroundColor: global.baseBackground100,
    },

    imageBanner:{
        width: '100%',
        height: '15%'
    },

    tabName:{
        justifyContent: 'center',
        width: '100%',
        height:'6%',
        backgroundColor: global.baseBlue100
    },

    tabText:{
        fontFamily: globalFont.chosenFont,
        paddingLeft: '3%',
        color: global.baseBackground100,
        fontSize: 18
    },

    buttonMenu:{
        justifyContent: 'center',
        alignItems: 'center',
        flex:1,
    },

    buttonWrapper:{
        backgroundColor: '#f2f2f2',
        width:'85%',
        height: 'auto',
        padding:'2%',
        justifyContent: 'center',
        borderRadius:15,
        marginTop:'3%',
        marginBottom:'3%',
    },

    buttonLabel:{
        fontFamily: globalFont.chosenFont,
        fontSize: 18
    },

    buttonDesc:{
        fontFamily: globalFont.chosenFont,
        fontSize: 12
    },

    buttons:{
        justifyContent: 'center',
        alignItems: 'center',
        alignSelf:'center',
        borderRadius: 10,
        width:'80%',
        height:'auto',
        backgroundColor: global.baseGold200,
        marginTop:'2%',
        marginBottom:'2%',
        paddingTop:'2%',
        paddingBottom:'2%'
    },

    buttonText:{
        fontFamily: globalFont.chosenFont,
        fontSize: 16,
    }
})