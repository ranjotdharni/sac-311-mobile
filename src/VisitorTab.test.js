import renderer from "react-test-renderer";
import VisitorTab from "./app/(tabs)/VisitorTab";


// example unit test to make sure the VisitorTab page renders
describe("VisitorTab", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<VisitorTab />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});


//   // it("has 4 children", () => {
//   //   const tree = renderer.create(<VisitorTab />).toJSON();
//   //   expect(tree.children.length).toBe(4);
//   // });