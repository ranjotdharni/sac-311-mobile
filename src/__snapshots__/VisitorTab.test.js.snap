// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisitorTab renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f5faff",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "paddingTop": "10%",
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#203d61",
        "height": "6%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#f5faff",
          "fontFamily": "jbm",
          "fontSize": 18,
          "paddingLeft": "3%",
        }
      }
    >
      Visitor Information
    </Text>
  </View>
  <Image
    source={1}
    style={
      {
        "height": "15%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#f2f2f2",
          "borderRadius": 15,
          "height": "auto",
          "justifyContent": "center",
          "marginBottom": "3%",
          "marginTop": "3%",
          "padding": "2%",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "jbm",
            "fontSize": 18,
          }
        }
      >
        Explore Sacramento
      </Text>
      <Text
        style={
          {
            "fontFamily": "jbm",
            "fontSize": 12,
          }
        }
      >
        Find out more about what there is to do in Sacramento.
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "rgba(253, 170, 31, 0.5)",
            "borderRadius": 10,
            "height": "auto",
            "justifyContent": "center",
            "marginBottom": "2%",
            "marginTop": "2%",
            "paddingBottom": "2%",
            "paddingTop": "2%",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "jbm",
              "fontSize": 16,
            }
          }
        >
          VisitCalifornia.com
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#f2f2f2",
          "borderRadius": 15,
          "height": "auto",
          "justifyContent": "center",
          "marginBottom": "3%",
          "marginTop": "3%",
          "padding": "2%",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "jbm",
            "fontSize": 18,
          }
        }
      >
        Book Your Stay
      </Text>
      <Text
        style={
          {
            "fontFamily": "jbm",
            "fontSize": 12,
          }
        }
      >
        Explore options booking a hotel, places to eat nearby, and more.
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "rgba(253, 170, 31, 0.5)",
            "borderRadius": 10,
            "height": "auto",
            "justifyContent": "center",
            "marginBottom": "2%",
            "marginTop": "2%",
            "paddingBottom": "2%",
            "paddingTop": "2%",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "jbm",
              "fontSize": 16,
            }
          }
        >
          VisitSacramento.com
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#f2f2f2",
          "borderRadius": 15,
          "height": "auto",
          "justifyContent": "center",
          "marginBottom": "3%",
          "marginTop": "3%",
          "padding": "2%",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          {
            "fontFamily": "jbm",
            "fontSize": 18,
          }
        }
      >
        Tripadvisor: Sacramento
      </Text>
      <Text
        style={
          {
            "fontFamily": "jbm",
            "fontSize": 12,
          }
        }
      >
        Find hotels, vacation rentals, browse travel forums, and more on Tripadvisor.
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "rgba(253, 170, 31, 0.5)",
            "borderRadius": 10,
            "height": "auto",
            "justifyContent": "center",
            "marginBottom": "2%",
            "marginTop": "2%",
            "paddingBottom": "2%",
            "paddingTop": "2%",
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "fontFamily": "jbm",
              "fontSize": 16,
            }
          }
        >
          Tripadvisor.com
        </Text>
      </View>
    </View>
  </View>
</View>
`;
